cg/services/pacbio/metrics/metrics_parser.py

```diff
@@ -1,72 +1,39 @@
-        self.report_dir = Path(smrt_cell_path, "statistics")
-        # For HiFi metrics
-        self.base_calling_report_file: Path = get_file_in_directory(
-            directory=self.report_dir, file_name=PacBioDirsAndFiles.BASECALLING_REPORT
-        )
-        self.hifi_metrics: HiFiMetrics = self.parse_report_to_model(
-            report_file=self.base_calling_report_file, data_model=HiFiMetrics
-        )
-        # For control metrics
-        self.control_report_file: Path = get_file_in_directory(
-            directory=self.report_dir, file_name=PacBioDirsAndFiles.CONTROL_REPORT
-        )
-        self.control_metrics: ControlMetrics = self.parse_report_to_model(
-            report_file=self.control_report_file, data_model=ControlMetrics
-        )
-        # For productivity metrics
-        self.loading_report_file: Path = get_file_in_directory(
-            directory=self.report_dir, file_name=PacBioDirsAndFiles.LOADING_REPORT
-        )
-        self.productivity_metrics: ProductivityMetrics = self.parse_report_to_model(
-            report_file=self.loading_report_file, data_model=ProductivityMetrics
-        )
-        # For polymerase metrics
-        self.raw_data_report_file: Path = get_file_in_directory(
-            directory=self.report_dir, file_name=PacBioDirsAndFiles.RAW_DATA_REPORT
-        )
-        self.polymerase_metrics: PolymeraseMetrics = self.parse_report_to_model(
-            report_file=self.raw_data_report_file, data_model=PolymeraseMetrics
-        )
-        # For SMRTlink datasets metrics
-        self.smrtlink_datasets_report_file: Path = get_file_in_directory(
-            directory=self.report_dir, file_name=PacBioDirsAndFiles.SMRTLINK_DATASETS_REPORT
-        )
-        self.smrtlink_datasets_metrics: SmrtlinkDatasetsMetrics = (
-            self.parse_smrtlink_datasets_file()
-        )
-
     @staticmethod
-    def parse_report_to_model(report_file: Path, data_model: Type[BaseMetrics]) -> BaseMetrics:
-        """Parse the metrics report to a data model."""
-        parsed_json: dict = ReadFile.read_file[FileFormat.JSON](file_path=report_file)
-        metrics: list[dict[str, Any]] = parsed_json.get("attributes")
-        data: dict = {report_field["id"]: report_field["value"] for report_field in metrics}
-        return data_model.model_validate(data, from_attributes=True)
+    def parse_metrics(smrt_cell_path: Path) -> PacBioMetrics:
+        """Return all the relevant PacBio metrics parsed in a single Pydantic object."""
+        report_dir = Path(smrt_cell_path, "statistics")
+        hifi_metrics: HiFiMetrics = parse_hifi_metrics(report_dir)
+        control_metrics: ControlMetrics = parse_control_metrics(report_dir)
+        productivity_metrics: ProductivityMetrics = parse_productivity_metrics(report_dir)
+        polymerase_metrics: PolymeraseMetrics = parse_polymerase_metrics(report_dir)
+        dataset_metrics: SmrtlinkDatasetsMetrics = parse_dataset_metrics(report_dir)
```

Comment:
Is it possible that some of the report files can be missing? What should happen if any of them are missing?

Reply:
An error would be raised in the `get_file_in_directory`:
```
FileNotFoundError: If the directory does not exist.
```
I think these kind of errors should be caught in an outer layer

Reply:
True, we raise custom exceptions there! For the error to be meaningful, I'd suggest raising a custom exception anyway in the utils module - feel free to ignore, but I think it would be an improvement